### ISSUE TITLE
API - Allow fractional sample rates

### ISSUE THREAD

**Issue by jwosty** [naudio/NAudio#755]
It would be really nice to be able to specify fractional sample rates for niche use cases (i.e. advanced resampling for visualization scenarios). 

There may be a way to do this without making a breaking change (i.e. add another property, `DoubleSampleRate`, and leave the exising `SampleRate`, which would update `DoubleSampleRate` when set). Alternatively it might be an acceptable good breaking change if breaking changes are ever considered.

**Comment by markheath** [comment]
Yes, this is an unfortunate artifact from `WaveFormat` being based on `WAVEFORMATEX`. I did look into whether I could introduce a new `AudioFormat` or `IAudioFormat` instead as part of NAudio 3 and have a clean break, but things got quite messy, and so I decided to back it out (maybe consider for NAudio 4 if that ever happens!).
